src/game_controller.py

```diff
@@ -75,45 +76,71 @@ def run_bot(self, pick_corpse: bool = False):
             if self._config.general["info_screenshots"]:
                 cv2.imwrite("./info_screenshots/info_could_not_recover_" + time.strftime("%Y%m%d_%H%M%S") + ".png", self.screen.grab())
             Logger.error(
-                f"{self._config.general['name']} could not recover from a max game length violation. Shutting down everything.")
+                f"{self._config.general['name']} could not recover from a max game length violation. Restarting the Game.")
+            if self._config.general["custom_message_hook"]:
+                messenger.send(msg=f"{self._config.general['name']}: got stuck and will now restart D2R")
+            if restart_game(self._config.general["d2r_path"]):
```

Comment:
is it possible to please move this to be an optional feature until we have the ability to select the correct Character from the hero selection screen? 

D2R restarted for me last night and upon restart botty played with my level 8 paladin as it is the first character I have on startup

I believe something like `if self._config.whatever["restart_config_name_whatever"]:` before this line + the addition of the config should suffice

Reply:
#403 Can you test this one? @FrancescoZini

Reply:
wow that was fast! :D yeah looks good, I've just left a comment for the case in which the character is down the list (if you have many) so you might have to scroll down

Reply:
Yeah... but... ahhh

Reply:
Yeah, @FrancescoZini, what you experienced is a bug in D2R. It should always default to the last character you used. In the last patch they said they fixed it, but.ive been able to reproduce it by logging in to an account across multiple PC's or vms.